compiler/src/dotty/tools/dotc/typer/Migrations.scala

```diff
@@ -139,14 +141,31 @@ trait Migrations:
       val rewriteMsg =
         if hasParentheses then
           Message.rewriteNotice("This code", mversion.patchFrom)
-        else
-          ""
-      report.errorOrMigrationWarning(
+        else ""
+      val message =
         em"""Implicit parameters should be provided with a `using` clause.$rewriteMsg
-            |To disable the warning, please use the following option: 
+            |To disable the warning, please use the following option:
             |  "-Wconf:msg=Implicit parameters should be provided with a `using` clause:s"
-            |""", 
-        pt.args.head.srcPos, mversion)
+            |"""
+      val withActions = message.withActions(
+          List(
+            CodeAction(
+              title = "Add `using` clause",
+              description = None,
+              patches = List(
+                ActionPatch(
+                  pt.args.head.startPos.sourcePos,
+                  "using "
+                )
+              )
+            )
+          )
+        )
```

Comment:
Could reduce parens by preferring colon syntax. I remember the impedance mismatch of supplying lists when there was always one patch and for action. Scala 2 had CodeAction.apply return a list of one action. If only Scala supported varargs...

Reply:
Fixed it up a bit

Reply:
I just saw a deeply nested stack of colons in the parser and that is also a challenge (for me) to read. It happens that the `args` on the preceding line is supposed to be the unused `args1`, so there was additional cognitive load.

https://github.com/scala/scala3/blob/main/compiler/src/dotty/tools/dotc/parsing/Parsers.scala#L1803